### PR TITLE
03 get api categories

### PR DESCRIPTION
**Pull request by alxjnr**

### CODE REVIEW

File: controllers/controllers.categories.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+const { fetchCategories } = require("../models/models.categories");
+
+exports.getCategories = (req, res) => {
+  fetchCategories().then((categories) => {
+    res.status(200).send({ categories });
```
Comment (lewispricey) [inline controllers/controllers.categories.js:5]: Good choice of key to return the data on!

File: db/connection.js [hunk 2]
```diff
@@ -1,12 +1,12 @@
...
+const ENV = process.env.NODE_ENV || "development";
 
-require('dotenv').config({
+require("dotenv").config({
   path: `${__dirname}/../.env.${ENV}`,
 });
 
 if (!process.env.PGDATABASE) {
-  throw new Error('PGDATABASE not set');
+  throw new Error("PGDATABASE not set");
```
Comment (lewispricey) [inline db/connection.js:9]: I like the use of throw new error here if there is no database available.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+beforeEach(() => {
+  return seed(data);
+});
+
+afterAll(() => {
+  connection.end();
+});
+
+describe("API Testing", () => {
+  describe("Categories", () => {
```
Comment (lewispricey) [inline __tests__/app.test.js]: Is your intention here of having a seperate describe block for each endpoint with a nested describe for the different request types? 

If so I like that this is all being organised together but the endpoint path ' /api/categories' may be a little clearer for the description than 'categories'

File: app.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+const express = require("express");
+const app = express();
+const { getCategories } = require("./controllers/controllers.categories");
+
+app.use(express.json());
```
Comment (lewispricey) [inline app.js]: Please could you remove this line until its required :)

File: models/models.categories.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+const db = require("../db/connection.js");
+
+exports.fetchCategories = () => {
+  return db.query(`SELECT * FROM categories`).then((res) => {
+    return res.rows;
```
Comment (lewispricey) [inline models/models.categories.js:5]: Good to see you're only returning the rows here 👍

**Review by lewispricey — APPROVED** [review]
Great work on this ticket Alex, I left a small comment about how we could make your test description a little clearer however its completely up to you whether you choose to make that change.

Please could you remove the express.json() from your app file until its required.

Overall really good stuff, please make the change above and feel free to merge 🚀

File: __tests__/app.test.js (hunk 3 above)
Comment (alxjnr) [reply __tests__/app.test.js]: this makes sense, will do that!

**Pull request merged**